daml/Marketplace/Distribution/Auction/Service.daml

```diff
@@ -0,0 +1,201 @@
+{-# LANGUAGE MultiWayIf #-}
+
+module Marketplace.Distribution.Auction.Service where
+
+import DA.Action (foldlA)
+import DA.Foldable (forA_)
+import DA.List (sortOn, mapAccumL)
+import DA.Finance.Asset (AssetDeposit)
+import DA.Finance.Types (Account, Asset, Id)
+import DA.Finance.Utils (fetchAndArchive)
+import Marketplace.Settlement qualified as Settlement
+import Marketplace.Distribution.Auction.Model qualified as Auction
+import Marketplace.Distribution.Auction.Model (Auction(..))
+import Marketplace.Distribution.Auction.Utils (dutchAuction, splitList, withdrawAllocation, updateBidStatus, generateSettlementInstruction)
+import Marketplace.Distribution.Bid.Service qualified as Bid
+import Marketplace.Distribution.Bid.Model qualified as Bid
+import Marketplace.Trading.AllocationAccountRule qualified as Allocation
+import Marketplace.Trading.AllocationAccountRule (AllocationAccountRule(..))
+
+type S = Service
+
+template Service
+  with
+    operator : Party
+    provider : Party
+    customer : Party
+  where
+    signatory operator, provider, customer
+
+    nonconsuming choice OfferBidder : ContractId Bid.Offer
```

Comment:
We should see if we can move this to the `Distributor` (currently `Auctioneer`) role and have the Bidding service be independent of any given auction. Would need some privacy-preserving way to invite bidders to a specific auction. Not something to do right now, but maybe leave a comment about it

Reply:
Done